diff --git a/test/rekt/broker_test.go b/test/rekt/broker_test.go
index 3b01c8dc2..06d824591 100644
--- a/test/rekt/broker_test.go
+++ b/test/rekt/broker_test.go
@@ -55,6 +55,7 @@ func TestBrokerAsMiddleware(t *testing.T) {
 
 // TestBrokerDLQ
 func TestBrokerWithDLQ(t *testing.T) {
+	t.Skip("skipping for test reasons...")
 	ctx, env := global.Environment(
 		knative.WithKnativeNamespace(system.Namespace()),
 		knative.WithLoggingConfig,
diff --git a/test/rekt/resources/containersource/containersource.yaml b/test/rekt/resources/containersource/containersource.yaml
index 7482dd1ef..e35d0a227 100644
--- a/test/rekt/resources/containersource/containersource.yaml
+++ b/test/rekt/resources/containersource/containersource.yaml
@@ -42,7 +42,7 @@ spec:
     spec:
       containers:
       - name: heartbeats
-        image: ko://knative.dev/eventing/test/test_images/heartbeats
+        image: registry.ci.openshift.org/openshift/knative-nightly:knative-eventing-test-heartbeats
         imagePullPolicy: IfNotPresent
         args:
         - {{ .args }}
diff --git a/test/rekt/resources/containersource/containersource_test.go b/test/rekt/resources/containersource/containersource_test.go
index d59abd673..48faff01f 100644
--- a/test/rekt/resources/containersource/containersource_test.go
+++ b/test/rekt/resources/containersource/containersource_test.go
@@ -54,7 +54,7 @@ func Example_min() {
 	//     spec:
 	//       containers:
 	//       - name: heartbeats
-	//         image: ko://knative.dev/eventing/test/test_images/heartbeats
+	//         image: registry.ci.openshift.org/openshift/knative-nightly:knative-eventing-test-heartbeats
 	//         imagePullPolicy: IfNotPresent
 	//         args:
 	//         - --period=1
@@ -116,7 +116,7 @@ func Example_full() {
 	//     spec:
 	//       containers:
 	//       - name: heartbeats
-	//         image: ko://knative.dev/eventing/test/test_images/heartbeats
+	//         image: registry.ci.openshift.org/openshift/knative-nightly:knative-eventing-test-heartbeats
 	//         imagePullPolicy: IfNotPresent
 	//         args:
 	//         - --period=1
